pkg/broker/broker.go

```diff
@@ -789,14 +769,41 @@ func (a AnsibleBroker) Unbind(
 		return nil, errors.New(errMsg)
 	}
 
+	params := make(apb.Parameters)
+	provExtCreds, err := a.dao.GetExtractedCredentials(instanceUUID.String())
+	if err != nil {
+		return nil, err
+	}
+	bindExtCreds, err := a.dao.GetExtractedCredentials(bindingUUID.String())
+	if err != nil && !client.IsKeyNotFound(err) {
+		return nil, err
+	}
+	// Add the credentials to the parameters so that an APB can choose what
+	// it would like to do.
+	params[provisionCredentialsKey] = provExtCreds.Credentials
+	if bindExtCreds != nil {
+		params[bindCredentialsKey] = bindExtCreds.Credentials
+	}
 	serviceInstance, err := a.getServiceInstance(instanceUUID)
 	if err != nil {
 		a.log.Debugf("Service instance with id %s does not exist", instanceUUID.String())
```

Comment:
Just noticing this; if this branch is taken it's a hard error and we need to return w/ error code. We'll potentially hit an NPE continuing with a nil `serviceInstance`.

This would also be quite bad; if someone is trying to unbind from a service that we don't know exists, we're in a real corrupt state.